### ISSUE TITLE
WebSockets in UI

### ISSUE THREAD

**Issue by kostyan5** [mitmproxy/mitmproxy#2929]
Are WebSockets supported in flow view (mitmproxy or mitmweb)?  I can only see websocket messages using mitmdump.  Using Mitmproxy: 4.0.0.

**Comment by kostyan5** [comment]
Is there a way to separate Websocket traffic (dump, logging) from everything else?  I would like to see websocket only traffic.

**Comment by Kriechi** [comment]
Yes, you can use `~websocket` to only match those flows. See https://www.mitmproxy.org/docs/latest/concepts-filters/

Right now our websocket implementation uses a single HTTP flow to hold all websocket traffic.
Please see the `websocket_*` related events: https://www.mitmproxy.org/docs/latest/addons-events/

**Comment by kostyan5** [comment]
I added the following to my config file and it didn’t do any filtering in mitmdump.  I still got all traffic.  I’m using it in transparent mode.

view_filter: ~websocket
save_stream_filter: ~websocket

**Comment by raamojirao** [comment]
@kostyan5 How to dump websocket messages using mitmdump?
